hoomd_benchmarks/write_gsd.py

```diff
@@ -0,0 +1,103 @@
+# Copyright (c) 2021-2023 The Regents of the University of Michigan
+# Part of HOOMD-blue, released under the BSD 3-Clause License.
+
+"""Benchmark GSD writes."""
+
+import hoomd
+import time
+from . import common
+from .configuration.hard_sphere import make_hard_sphere_configuration
+
+DEFAULT_BANDWIDTH = 'false'
+DEFAULT_MAXIMUM_WRITE_BUFFER_SIZE = 64 * 1024 * 1024
+
+
+class GSD(common.Benchmark):
+    """Base class GSD benchmark.
+
+    Args:
+        kwargs: Keyword arguments accepted by ``Benchmark.__init__``
+
+    Derived classes should set override `make_write_gsd` to implement
+    different options.
+
+    See Also:
+        `common.Benchmark`
+    """
+
+    def __init__(self,
+                 bandwidth=DEFAULT_BANDWIDTH,
+                 maximum_write_buffer_size=DEFAULT_MAXIMUM_WRITE_BUFFER_SIZE,
+                 **kwargs):
+        self.bandwidth = bandwidth
+        self.maximum_write_buffer_size = maximum_write_buffer_size
+
+        super().__init__(**kwargs)
+
+        if bandwidth:
+            self.units = 'MiB/s'
+
+    @staticmethod
+    def make_argument_parser():
+        """Make an ArgumentParser instance for benchmark options."""
+        parser = common.Benchmark.make_argument_parser()
+        parser.add_argument('--bandwidth',
+                            action='store_true',
```

Comment:
```suggestion
                            action='store_false',
```
Should this be `store_false`, since the default for the bandwidth argument appears to be `False`?

Reply:
No, `store_true` is correct. Adding `--bandwidth` to the command line should display the results in MiB/s, not disable it:
```
▶ python3 -m hoomd_benchmarks.write_gsd --device CPU -v            
Using existing initial_configuration_cache/hard_sphere_64000_1.0_3.gsd
Running GSD benchmark
.. warming up for 1000 steps
.. running for 1000 steps 1 time(s)
.. 1411.3981149745082 time steps per second
1411.3981149745082
▶ python3 -m hoomd_benchmarks.write_gsd --device CPU --bandwidth -v 
Using existing initial_configuration_cache/hard_sphere_64000_1.0_3.gsd
Running GSD benchmark
.. warming up for 1000 steps
.. running for 1000 steps 1 time(s)
.. 1031.269418128387 MiB/s
1031.269418128387
```